dss-user-manager/src/main/java/com/webank/wedatasphpere/dss/user/service/impl/SchedulisCommand.java

```diff
@@ -0,0 +1,128 @@
+
+import cn.hutool.crypto.SecureUtil;
+import cn.hutool.crypto.symmetric.DES;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * @author litongtong
+ * @date 2021/1/11
+ */
+public class SchedulisCommand extends AbsCommand {
+
+    private static final Logger logger = LoggerFactory.getLogger(SchedulisCommand.class);
+
+    @Override
+    public String authorization(AuthorizationBody body) {
+        String username = body.getUsername();
+        logger.info("开始新增调度用户：{}", username);
+        String password = body.getPassword();
+        String encryptionPwd = getEncryptionPwd(username, password);
+        Connection connection;
+        Statement stmt;
+        try {
+            logger.info("开始插入ctyun_user表");
+            connection = getConnection();
+            stmt = connection.createStatement();
+            String sql = "INSERT INTO `ctyun_user` (`id`,`name`,`username`,`email`,`password`,`work_order_item_config`) VALUES (?,?,?,?,?,NULL) ON DUPLICATE KEY UPDATE `password` = ?";
+            PreparedStatement statement = connection.prepareCall(sql);
+            statement.setString(1, username);
+            statement.setString(2, username);
+            statement.setString(3, username);
+            statement.setString(4, username);
+            statement.setString(5, encryptionPwd);
+            statement.setString(6, encryptionPwd);
+            statement.executeUpdate();
+            stmt.close();
+            connection.close();
+            logger.info("完成插入ctyun_user表");
+        } catch (SQLException e) {
+            logger.error(e.getMessage());
+        }
+        logger.info("开始调用接口新增schedulis用户");
+        addSchedulisUser(username, password);
+        logger.info("结束调用接口新增schedulis用户");
+        return Command.SUCCESS;
+    }
+
+    private static Connection getConnection() {
+        try {
+            //注册数据库的驱动
+            Class.forName("com.mysql.jdbc.Driver");
+            //获取数据库连接（里面内容依次是：主机名和端口、用户名、密码）
+            String url = DSSUserManagerConfig.BDP_SERVER_MYBATIS_DATASOURCE_URL;
+            String user = DSSUserManagerConfig.BDP_SERVER_MYBATIS_DATASOURCE_USERNAME;
+            String password = DSSUserManagerConfig.BDP_SERVER_MYBATIS_DATASOURCE_PASSWORD;
+            return DriverManager.getConnection(url, user, password);
+        } catch (Exception e) {
+            logger.error(e.getMessage());
+        }
+        return null;
+    }
+
+    private boolean addSchedulisUser(String username, String password) {
+        RestTemplate restTemplate = new RestTemplate();
+        String schedulisUrl = DSSUserManagerConfig.SCHEDULER_ADDRESS;
+        String url = new StringBuilder().append(schedulisUrl)
+            .append("/luban/schedule/system")
```

Comment:
this url suggest to modify,  make the url configurable

Reply:
- [ ] 

- [ ] We have used parameters: DSSUserManagerConfig.KERBEROS_ ENABLE_ Switch controls it

Reply:
Thank you. It has been revised